### DISCUSSION TITLE
Adjust default styling in header

### DISCUSSION THREAD

**Discussion by rbraeunlich** [mjmlio/mjml#2874] [Q&A] [answered]
Hi everybody,
I currently use MJML to generate some parts of my emails and then embed it. I know that MJML is not supposed to be embedded but still, this is how it currently works for me. However, I have a strange interference when it comes to the styling.
As far as I could see, MJML always generates this styling:
```
    body {
      margin: 0;
      padding: 0;
      -webkit-text-size-adjust: 100%;
      -ms-text-size-adjust: 100%;
    }

    table,
    td {
      border-collapse: collapse;
      mso-table-lspace: 0pt;
      mso-table-rspace: 0pt;
    }

    img {
      border: 0;
      height: auto;
      line-height: 100%;
      outline: none;
      text-decoration: none;
      -ms-interpolation-mode: bicubic;
    }

    p {
      display: block;
      margin: 13px 0;
    }</style>
```
even when using a minimal template (e.g. see HTML here https://mjml.io/try-it-live). The `p` definition makes some elements that should be hidden on Desktop appear in Outlook desktop (those elements were added afterwards not via MJML). Therefore, I was wondering if there is any way to influence this general styling or to remove it somehow.
Thanks for your help.

**Comment by iRyusa** [comment, marked as answer]
It's part of the [Skeleton](https://github.com/mjmlio/mjml/blob/master/packages/mjml-core/src/helpers/skeleton.js) which can be override as part of `mjml2html` options. But I strongly advise to not do this as those reset ensure email compatibility. 
Just use `!important` maybe ?

**Reply by rbraeunlich** [reply]
Hi @iRyusa,
thank you for the quick answer. Unfortunately, `!important` isn't an option since I can directly influence the other CSS.
Could you tell me which parameter/option I have to use to overwrite the skeleton? I cannot find anything in the documentation.

**Reply by iRyusa** [reply]
It's an option in `mjml2html` as `skeleton` which can be a js module or a function directly. You can provide it in your `.mjmlconfig` too but keep in mind I wouldn't really go this way as you would miss some fix in the future. 

```
mjml2html(input, { 
  skeleton: (options) => `your own html + options` 
})
```

It's not really documented as it's most likely used for internal stuff
